src/main/java/net/glowstone/entity/GlowTntPrimed.java

```diff
@@ -22,8 +23,28 @@
     @Getter
     @Setter
     private int fuseTicks;
+    @Setter
     private Entity source;
 
+    /**
+     * Get the player that ignited the TNT.
+     *
+     * @return Player that ignited the TNT
+     */
+    public GlowPlayer getPlayer() {
+        Entity source = getSource();
+        return (source instanceof GlowPlayer) ? (GlowPlayer) source : null;
+    }
+
+    /**
+     * Set the player that ignited the TNT.
+     *
+     * @param player Player that ignited the TNT
+     */
+    public void setPlayer(Player player) {
```

Comment:
Can you remove this setter and just use `setSource` with the player? I don't think having a separate method here is necessary.